### DISCUSSION TITLE
Change byte range formatting

### DISCUSSION THREAD

**Discussion by hackerfactor** [hackerfactor/SEAL#6] [Ideas]
As discussed in #4, my initial definition for the byte range parameter (`b=`) contains some ambiguities:

- Hyphens are used for both the byte range and for math. E.g., `b=p-(s-50)`.  The recommendation is to use hyphens for math and tilda for range: `b=p~s-50`. This makes the parenthesis unnecessary.
- There should be a variable for the beginning and end of the file: `b` and `e`. Such as `b=b~s,s~e` for the entire file (excluding the signature).
- The `s` literal has two meanings. On the left-side of the range, it refers to the start of the signature. One the right-side, it refers to the end of the signature. The proposal is to use two different literals for the start and end of the signature. This hasn't been resolved yet.

**Comment by named-dodo** [comment]
The current updated format plan as far as i am aware:
Example string is `b="F~S-100,S-50~s,s+10~f"`
1. Split on `,` to get an array of ranges: `F~S-100` / `S-50~s` / `s+10~f`
2. Split on `~` to get the begin and end part of the range:  `s+10~f` => `s+10` / `f`
3. A range part _must_ start with a letter to set a base value:
  `F` for begin of file
  `f` for end of file 
 `S` for begin of signature 
 `s` for end of signature 
 `P` for begin of previous signature
 `p` for end of previous signature
4. Each range part can optionally have an offset: a `+` or `-` followed by a decimal number of bytes.

The default value would be `b="F~S,s~f"`, meaning _from begin of file to begin of signature, from end of signature to end of file._
This means that everything except for the signature is signed. (as you can't sign the signature itself.)

**Reply by hackerfactor** [reply]
I like that a lot. For me, it's as readable and makes sense.

**Comment by hackerfactor** [comment]
Specs updated to match.
